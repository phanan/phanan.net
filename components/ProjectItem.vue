<template>
  <li>
    <a :href="project.url" rel="me">{{ project.name }}
      <span class="url">{{ fqdn }}</span>
    </a>
  </li>
</template>

<script>
import config from '~/phanan.config'
import isLinkObject from '~/validators/isLinkObject'

export default {
  name: 'Project-Item',

  props: {
    project: {
      type: Object,
      required: true,
      validator: isLinkObject
    }
  },

  computed: {
    fqdn () {
      return this.project.url.match(config.URL_REGEX)[1]
    }
  }
}
</script>


<style lang="scss" scoped>
@import "../assets/scss/_mixins";

li {
  width: 100%;
  margin: 8px 0;

  a {
    &:hover {
      color: rgba(255, 255, 255, .2);

      .url {
        color: #fff;
        margin-left: 0;

        &:hover {
          color: rgba(255, 255, 255, .4);
        }
      }
    }
  }

  .url {
    color: rgba(255, 255, 255, .2);
    transition: color .5s, margin-left .5s;
    margin-left: -1200px;

    &::before {
      @include slash;
    }

    @media only screen and (max-width: 640px) {
      display: none;
    }
  }

  @media only screen and (max-width: 640px) {
    border-bottom: 1px solid #111;

    &:last-child {
        border-bottom: none;
    }
  }
}

</style>
