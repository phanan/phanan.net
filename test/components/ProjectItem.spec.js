import { renderToString } from '@vue/server-test-utils'
import ProjectItem from '~/components/ProjectItem'

describe('ProjectItem.vue', () => {
  it('matches snapshot', () => {
    expect(renderToString(ProjectItem, {
      propsData: {
        project: {
          name: 'Foo Bar',
          url: 'https://foo.bar'
        }
      }
    })).toMatchSnapshot()
  })
})
