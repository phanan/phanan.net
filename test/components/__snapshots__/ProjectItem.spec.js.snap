// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem.vue matches snapshot 1`] = `
<li>
  <a href="https://foo.bar" rel="me">Foo Bar
    <span class="url">foo.bar</span></a>
</li>
`;
